treeherder/webapp/api/intermittents_view.py

```diff
@@ -0,0 +1,103 @@
+import re
+
+from django.db.models import Count
+from django.db.models.functions import TruncDate
+from rest_framework import generics
+
+from pagination import CustomPagePagination
+from treeherder.config.settings import OPTION_COLLECTION_HASH_MAP
+from treeherder.etl.common import (get_end_of_day,
+                                   get_tree)
+from treeherder.model.models import (BugJobMap,
+                                     Job,
+                                     Push)
+
+from .serializers import (FailureCount,
+                          FailuresByBugSerializer,
+                          FailuresSerializer)
+
+
+class Failures(generics.ListAPIView):
+    """ List of intermittent failures by date range and tree (project name) """
+
+    serializer_class = FailuresSerializer
+    pagination_class = CustomPagePagination
+
+    def get_queryset(self):
+        startday = self.request.query_params.get('startday').encode('utf-8') + ' 00:00:00'
+        endday = get_end_of_day(self.request.query_params.get('endday').encode('utf-8'))
+        tree = get_tree(self.request.query_params.get('tree').encode('utf-8'))
+
+        queryset = BugJobMap.objects.filter(job__repository_id__in=tree, job__push__time__range=(startday, endday),
```

Comment:
This seems like it would be more read-able if it was formatted like:

    queryset = BugJobMap.objects.filter(
        job__repository_id__in=tree, 
        job__push__time__range=(startday, endday),
        job__failure_classification__id=4
        ).select_related('push').values('bug_id').annotate(bug_count=Count('job_id')).values(
            'bug_id', 'bug_count').order_by('-bug_count')

or something along those lines.  I must admit I'm not a fan of using the ``\`` for line continuation when you've got params you can break on.  But maybe that's just me.  :)

Reply:
Makes sense - will do.